### PR TITLE
Updated archive sidebar list with self-generating code

### LINKED ISSUE #144: Make archive year/month list self generating
it shouldn't be static

### PR DESCRIPTION
**Pull request by pmahnke**
## Done

- Updated archive sidebar list with self-generating code
- Added a custom filter for month abbreviations ‘monthname’

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: [a monthly archive page](http://0.0.0.0:8023/archives/2017/01/)

## Notes

* Data goes back further, but not with 12-month coverage
    * Should we be checking for any months that are missing data?
* Obviously, there is no design here, just my guess at something concise

## Issue / Card

Fixes #144

## Screenshots

image

### CODE REVIEW

File: templates/archive-list.html [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-    <li class="p-list__item"><a href="/archives/2017/02">February 2017</a></li>
-    <li class="p-list__item"><a href="/archives/2017/01">January 2017</a></li>
-    <li class="p-list__item"><a href="/archives/2016/12">December 2016</a></li>
-  </ul>
\ No newline at end of file
+  {% for year in range(today.year,2006,-1) %}
+  <li class="p-list__item">{{ year }}
+    <ul class="p-inline-list--middot">
+    {% for month in range(1, 13, 1) %}
+      {% if today.year == year and month > today.month %}
```
Comment (Lukewh) [inline templates/archive-list.html]: Could this not be `{% if today.year != year or month < today.month %}` and then remove the `{% else %}`?

File: templates/archive-list.html [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
+      {% else %}
+      <li class="p-inline-list__item"><a href="/archives/{{ year }}/{% if month < 10 %}0{% endif %}{{ month }}">{{ month | monthname }}</a></li>
```
Comment (Lukewh) [inline templates/archive-list.html]: Did we decide in a dev catchup that lines like this should be indented?

**Review by Lukewh — CHANGES_REQUESTED** [review]
One question/change and another question.

File: templates/archive-list.html (hunk 2 above)
Comment (WillMoggridge) [reply templates/archive-list.html]: Yeah, we did indeed

File: templates/archive-list.html (hunk 1 above)
Comment (pmahnke) [reply templates/archive-list.html]: @Lukewh not really... it is trying to NOT print Feb 2018 as it hasn't happened... I think your's isn't the inverse  - however this is - {% if today.year != year or (today.year == year and month <= today.month) %} but more complicated

Comment (Lukewh) [reply templates/archive-list.html]: ah, yeah you're right. I prefer the more complicated approach but I'm okay with you not changing it if you don't want to.

File: templates/archive-list.html (hunk 2 above)
Comment (pmahnke) [reply templates/archive-list.html]: fixed

File: templates/archive-list.html (hunk 1 above)
Comment (pmahnke) [reply templates/archive-list.html]: fixed

**Review by Lukewh — APPROVED** [review]

**Comment by pmahnke** [comment]
@frankiegranato here is a good page - http://insights.ubuntu.com-pr-145.run.demo.haus/archives/2016/12/

File: templates/archive-list.html [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
-    <li class="p-list__item"><a href="/archives/2017/05">May 2017</a></li>
-    <li class="p-list__item"><a href="/archives/2017/04">April 2017</a></li>
-    <li class="p-list__item"><a href="/archives/2017/03">March 2017</a></li>
-    <li class="p-list__item"><a href="/archives/2017/02">February 2017</a></li>
-    <li class="p-list__item"><a href="/archives/2017/01">January 2017</a></li>
-    <li class="p-list__item"><a href="/archives/2016/12">December 2016</a></li>
-  </ul>
\ No newline at end of file
+  {% for year in range(today.year,2006,-1) %}
+  <li class="p-list__item"><h5><a class="p-link--soft" href="/archives/{{ year }}/">{{ year }}</a></h5>
```
Comment (anthonydillon) [inline templates/archive-list.html]: Currently archives do not support just years.

File: templates/archive-list.html [hunk 4]
```diff
@@ -1,16 +1,14 @@
...
+    <ul class="p-inline-list--middot">
+    {% for month in range(1, 13, 1) %}
+      {# need to skip months in the current year that haven't occured #}
+      {% if today.year != year or (today.year == year and month <= today.month) %}
```
Comment (anthonydillon) [inline templates/archive-list.html]: This seems like a lot of logic for a template. I think this should be done in the model and the template just output dumbly.

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Few comments in code

**Comment by frankiegranato** [comment]
I can't see the demo. I see a error page: IndexError: list index out of range :-(

**Review by anthonydillon — APPROVED** [review]

**Pull request merged**